src/Lecture1.hs

```diff
@@ -81,7 +86,17 @@ and 1 is the smallest, and 7 - 1 = 6.
 Try to use local variables (either let-in or where) to implement this
 function.
 -}
-minmax x y z = error "TODO"
+minmax :: Int -> Int -> Int -> Int
+minmax x y z = 
+   let  (min_num, max_num) = if x <= y then (x, y) else (y, x)
+   in
+    if z <= min_num
+       then max_num - z
+       else if z >= max_num
+               then z - min_num
+               else max_num - min_num
```

Comment:
You can implement this function much shorter by using standard Haskell functions `min` and `max`:

```haskell
λ: max 7 3
7
λ: min 7 3
3
```